pkg/processor/k8sprocessor/pod_association.go

```diff
@@ -32,9 +32,16 @@ import (
 func extractPodID(ctx context.Context, attrs pdata.AttributeMap, associations []kube.Association) (podIdentifierKey string, podIdentifierValue kube.PodIdentifier) {
 	hostname := stringAttributeFromMap(attrs, conventions.AttributeHostName)
 	var connectionIP kube.PodIdentifier
-	if c, ok := client.FromContext(ctx); ok {
-		connectionIP = kube.PodIdentifier(c.IP)
+
+	if c := client.FromContext(ctx); c.Addr != nil {
+		ip, _, err := net.SplitHostPort(c.Addr.String())
```

Comment:
In what circumstances port might be available in the context?

Reply:
I'm not too familiar with how tracing workflow works 😕 The UTs pass, if we can add an additional set of tests that would make us more comfortable I'm all in on that.

Reply:
I think this is largely unnecessary at this point (see the comment above). The `client.Info` should already take care of extracting the address. I believe we might skip the extra processing and do `c.Addr.String()` without additional checks (though they do not seem to do harm, except for extra processing time, probably miniscule)